tests/app/main/test_frameworks.py

```diff
@@ -609,6 +624,58 @@ def test_upload_message_if_agreement_is_not_returned(self, data_api_client):
             assert_not_in(u'Document uploaded', data)
             assert_not_in(u'Your document has been uploaded', data)
 
+    def test_loads_contract_start_page_if_framework_agreement_version_exists(self, data_api_client):
+        with self.app.test_client():
+            self.login()
+
+            data_api_client.get_framework.return_value = get_g_cloud_8()
+            data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(
+                on_framework=True)
+
+            res = self.client.get("/suppliers/frameworks/g-cloud-8/agreement")
+            data = res.get_data(as_text=True)
+
+            assert res.status_code == 200
+            assert u'Return your signed signature page' in data
+            assert u'Send document to CCS' not in data
+
+    def test_two_lots_passed_on_contract_start_page(self, data_api_client):
+        with self.app.test_client():
+            self.login()
+
+            data_api_client.get_framework.return_value = get_g_cloud_8()
+            data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(
+                on_framework=True)
+            data_api_client.find_draft_services.return_value = {
+                'services': [
+                    {'lotSlug': 'saas', 'status': 'submitted'},
+                    {'lotSlug': 'scs', 'status': 'submitted'}
+                ]
+            }
+            expected_lots_and_statuses = [
+                ('Software as a Service', 'Pass'),
+                ('Platform as a Service', 'No application'),
+                ('Infrastructure as a Service', 'No application'),
+                ('Specialist Cloud Services', 'Pass'),
+            ]
+
+            res = self.client.get("/suppliers/frameworks/g-cloud-8/agreement")
+            doc = html.fromstring(res.get_data(as_text=True))
+
+            assert res.status_code == 200
+
+            lots_and_statuses = []
+            lot_table_rows = doc.xpath('//*[@id="content"]//table/tbody/tr')
```

Comment:
Incidental sidenote: did you know about xpath's `normalize-space()`? https://developer.mozilla.org/en-US/docs/Web/XPath/Functions/normalize-space

You can see how I've used it in `test_briefs.py`

Can be used to write robustify your tests and quicker to work with than external `strip()` monkeying...